File: cases/Python/bit_flip.py
# Flip all bits
def flip(bits):
    if (len(bits) > 0):
      if (bits[0] == 0):
        return [1, flip(bits[1])]
      else:
        return [0, flip(bits[1])]
    else:
      return []

def main():
  
  # Immutable list of 32 bits (zeroes)
  bits = [0, [0, [0, [0, [0, [0, [0, [0, [0, [0, [0, [0, [0, [0, [0, [0, [0, [0, [0, [0, [0, [0, [0, [0, [0, [0, [0, [0, [0, [0, [0, [0,[]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]];

  lim = 2**20
  for i in range(lim):
    bits = flip(bits)
    #print(bits)

main()

File: cases/Haskell/bit_flip.hs
-- Flip all bits
flipBits :: [Int] -> [Int]
flipBits [] = []
flipBits (0:xs) = 1:flipBits(xs)
flipBits (_:xs) = 0:flipBits(xs)

main = do
  let bits = replicate 32 0-- Immutable list of 32 bits (zeroes)
  let lim = 2^20
  print (iterate (flipBits) (bits) !! lim)
